### PR TITLE
feat: add updatedAt index + server-side updatedSince/repo filtering to task-store

### PR DESCRIPTION
**Pull request by dodizzle**
## Summary
- Add `@@index([updatedAt])` to `Task` and `PullRequest` in `task-store/prisma/schema.prisma`, with a migration
- Add server-side `updatedSince` filtering to `GET /tasks` and `GET /prs`, wired through `TaskService.list()` / `PullRequestService.list()` as `where.updatedAt = { gte: new Date(updatedSince) }`
- Regenerate `task-store/openapi.json`, `lib/task-store-types.ts`, `mcp-server/src/generated-tools.ts`, `docs/mcp-tools.md`, and refresh `docs/task-store.md`

This is part of fixing production 503s on the shipwright-metrics dashboard, caused by `metrics/src/lib/task-store-client.ts` paging through the entire task-store history on every dashboard refresh. `updatedAt >= from` is a conservative pre-filter (not a precise sync anchor) — see `planning/task-store-date-filtering/PLAN.md` for the full root-cause and design rationale.

## Acceptance Criteria

| # | Criterion | Status |
|---|-----------|--------|
| 1 | `GET /tasks?updatedSince=<iso>` and `GET /prs?updatedSince=<iso>` exclude rows with `updatedAt` before the cutoff; omitting the param preserves current (unfiltered) behavior | MET |
| 2 | `repo` continues to filter correctly, unchanged | MET |
| 3 | Migration applies cleanly via `prisma migrate dev` with no data loss; openapi.json/generated types/mcp-docs regenerated and biome-formatted | MET |
| 4 | Unit tests (`task-service.unit.test.ts`/`pull-request-service.unit.test.ts`), integration tests against the real test DB (`tasks.integration.test.ts`/`pull-request.integration.test.ts`), and OpenAPI smoke tests (`routes/tasks.openapi.smoke.test.ts`/`routes/prs.openapi.smoke.test.ts`) all cover `updatedSince`. No existing tests retired. | MET |

## Test Plan
- `bun test` in `task-store/` — 309 pass, 93 skip (integration tests require `DATABASE_URL_SHIPWRIGHT_TASK_STORE_TEST`, provisioned in CI), 0 fail
- `bun run typecheck` — clean across all workspaces
- `bunx biome lint .` — clean
- [x] Pre-commit checks passing

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: task-store/src/task-service.ts [hunk 1]
```diff
@@ -106,6 +113,9 @@ export class TaskService implements TaskServiceLike {
     if (filters.claimedBy) where.claimedBy = filters.claimedBy;
     if (filters.pr !== undefined) where.pr = filters.pr;
     if (filters.branch !== undefined) where.branch = filters.branch;
+    if (filters.updatedSince) {
+      where.updatedAt = { gte: new Date(filters.updatedSince) };
```
Comment (dodizzle) [inline task-store/src/task-service.ts]: `filters.updatedSince` is a raw caller-supplied query string passed straight to `new Date()` with no format validation anywhere in the stack — the Zod schema is bare `z.string().optional()` (`openapi-schemas.ts:395`), the route handler passes it through unchecked (`routes/tasks.ts:510`), and there's no `isNaN(date.getTime())` check here. An invalid value like `?updatedSince=not-a-date` produces `Invalid Date`, which Prisma will throw on when serializing the query — caught only by `app.ts`'s generic 500 handler, not the `BadRequestError` → 400 path used for `repo` (`validateRepo`) and `prNumber` (`Number.isInteger`) in this same file. Worth validating and throwing `BadRequestError` on parse failure to match the existing pattern. (Mirrored in `pull-request-service.ts:119`.)

File: task-store/src/tasks.integration.test.ts [hunk 2]
```diff
@@ -737,4 +737,91 @@ describeOrSkip("Task store schema (integration)", () => {
     expect(result.tasks.map((t) => t.title)).toEqual(["Own task"]);
     expect(result.total).toBe(1);
   });
+
+  // ─── TaskService.list() updatedSince filter ─────────────────────────────────
+
+  it("list({ updatedSince }) excludes tasks whose updatedAt is before the cutoff", async () => {
```
Comment (dodizzle) [inline task-store/src/tasks.integration.test.ts:743]: AC4 claims full `updatedSince` test coverage, but none of the six touched test files (unit/integration/smoke across tasks and PRs) exercise an invalid `updatedSince` string, and none test the exact-boundary case (`updatedAt == updatedSince`, verifying `gte` inclusivity). All integration cutoffs sit strictly between an old and new timestamp. Given the route has zero input validation right now, this is exactly the gap where that would've been caught — worth adding a boundary case here and an invalid-input case at the route/service layer.

**Review by dodizzle — COMMENTED** [review]
Verdict: COMMENT — `updatedSince` is passed to `new Date()` with no validation (unlike `repo`/`prNumber` in the same handlers), so a malformed value surfaces as an alert-triggering 500 instead of a 400 — and no test covers that case or the exact `gte` boundary despite AC4 claiming full coverage. Core feature (index + filter + repo composition) is otherwise clean and well-tested. Details in the two inline comments.

**Review by dodizzle — COMMENTED** [review]
Verdict: APPROVE — Both prior findings are resolved: `updatedSince` now goes through a `parseUpdatedSince()` helper that throws `BadRequestError` (mapped to 400) on an unparseable value in both `TaskService.list()` and `PullRequestService.list()`, matching the existing `repo`/`prNumber` validation pattern. New unit/integration/smoke tests cover the invalid-input 400 path and the exact `updatedAt == updatedSince` gte-boundary case. Tests, typecheck, and lint all pass at head. Posting as COMMENT since GitHub blocks self-APPROVE via the API — the recorded verdict is APPROVE.
